tools/str_tools.go

```diff
@@ -46,6 +46,22 @@ func QuotedFields(s string) []string {
 	return out
 }
 
+// ShellQuote returns a copied string slice where each element is quoted
+// suitably for sh.
+func ShellQuote(strs []string) []string {
+	dup := make([]string, 0, len(strs))
+
+	for _, str := range strs {
+		// Quote anything that looks slightly complicated.
+		if shellWordRe.FindStringIndex(str) == nil {
+			dup = append(dup, "'" + strings.Replace(str, "'", "'\\''", -1) + "'")
```

Comment:
Nit; `go fmt` encourages that there not be whitespace between `+` and either of its operands.

To fix this, you can either run `make fmt`:

```ShellSession
$ make fmt
tools/str_tools.go
```

Which produces the following diff:

```diff
$ git diff
diff --git a/tools/str_tools.go b/tools/str_tools.go
index 50bba537..14f6debe 100644
--- a/tools/str_tools.go
+++ b/tools/str_tools.go
@@ -54,7 +54,7 @@ func ShellQuote(strs []string) []string {
        for _, str := range strs {
                // Quote anything that looks slightly complicated.
                if shellWordRe.FindStringIndex(str) == nil {
-                       dup = append(dup, "'" + strings.Replace(str, "'", "'\\''", -1) + "'")
+                       dup = append(dup, "'"+strings.Replace(str, "'", "'\\''", -1)+"'")
                } else {
                        dup = append(dup, str)
                }
```

Or, you can run `goimports` (which acts like `go fmt`, but also imports packages that are referenced but not imported) before saving in your editor. FWIW, I use Vim and use https://github.com/fatih/vim-go, which runs a command of your choice on buffer save, (mine is configured [here](https://github.com/ttaylorr/dotfiles/blob/7b0323b5e34bcb7ab623e98be98a1373d5d18d17/vim/.vimrc#L55-L56)).

Reply:
I thought I had run it through `gofmt` using Vim, but apparently I forgot.  Will fix.